### PR TITLE
Image download and related updates to support rich push from the SDK

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description

Added rich push support to android SDK

# Check List

- [x] Are you changing anything with the public API? -- NO
- [x] Are your changes backwards compatible with previous SDK Versions? -- YES
- [x] Have you tested this change on real device? -- YES
- [ ] Have you added unit test coverage for your changes?
- [x] Have you verified that your changes are compatible with all Android versions the SDK currently supports? -- Yes, by using the NotificationBuilderCompat class we should be covered.

## Changelog / Code Overview

- Updates the push-fcm package to support rich push.
- Added image download and apply for klaviyo messages that have media attached

## Test Plan

- Tested in a real app to confirm image display both with a locally generated notification and a remote message.

### CODE REVIEW

File: sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoNotification.kt [hunk 1]
```diff
@@ -124,7 +139,43 @@ class KlaviyoNotification(private val message: RemoteMessage) {
             .setNumber(message.notificationCount)
             .setPriority(message.notificationPriority)
             .setAutoCancel(true)
-            .build()
+
+    private fun URL.applyToNotification(builder: NotificationCompat.Builder) {
+        val executor = Executors.newCachedThreadPool()
```
Comment (evan-masseau) [inline sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoNotification.kt:144]: Non-blocking todo: I suppose we ought to have a unit test covering this case

File: sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoNotification.kt [hunk 2]
```diff
@@ -124,7 +139,43 @@ class KlaviyoNotification(private val message: RemoteMessage) {
             .setNumber(message.notificationCount)
             .setPriority(message.notificationPriority)
             .setAutoCancel(true)
-            .build()
+
+    private fun URL.applyToNotification(builder: NotificationCompat.Builder) {
```
Comment (evan-masseau) [inline sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoNotification.kt:143]: If we did want to expose a helper method for downloading and applying the image, all we'd have to do is make this extension method public @ajaysubra circling back to yesterday's conversation.

**Review by ajaysubra — APPROVED** [review]

**Review by ajaysubra — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -339,6 +341,9 @@ through Klaviyo to make sure that the URI shows up in the handler you implemente
 For more in-depth information on deep linking,
 refer to [android developer documentation](https://developer.android.com/training/app-links/deep-linking).
 
+### Rich push notification 
+
+In order for rich push notifications to work on your app, your app needs to be on the Klaviyo SDK version 1.2.0.
```
Comment (jordan-griffin) [inline README.md:346]: ```your app needs to be on or above version 1.2.0 of the Klaviyo SDK``` I'd say here I guess

File: sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoNotification.kt [hunk 4]
```diff
@@ -114,7 +129,7 @@ class KlaviyoNotification(private val message: RemoteMessage) {
      * @param context
      * @return [Notification] to display
      */
-    private fun buildNotification(context: Context): Notification =
+    private fun buildNotification(context: Context): NotificationCompat.Builder =
```
Comment (jordan-griffin) [inline sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoNotification.kt:132]: nit: I feel like this kinda changes the docstring tied to the function too. We're no longer returning a `Notification` instance but a builder

Comment (evan-masseau) [reply sdk/push-fcm/src/main/java/com/klaviyo/pushFcm/KlaviyoNotification.kt:132]: oh most definitely

File: README.md (hunk 3 above)
Comment (ajaysubra) [reply README.md]: Oops I missed adding on or above. Good catch. Updated.

**Review by ajaysubra — APPROVED** [review]

**Review by jordan-griffin — APPROVED** [review]

**Review by ajaysubra — APPROVED** [review]

**Review by kennyklaviyo — APPROVED** [review]

**Pull request merged**